generators/client/templates/angular/src/main/webapp/app/account/settings/settings.component.ts.ejs

```diff
@@ -53,64 +52,53 @@ export class SettingsComponent implements OnInit {
         private languageService: JhiLanguageService,
         private languageHelper: JhiLanguageHelper<% } %>
     ) {
+        <%_ if (enableTranslation) { _%>
+        this.languages = this.languageHelper.getAll();
+        <%_ } _%>
     }
 
-    ngOnInit() {
+    ngOnInit(): void {
         this.accountService.identity().subscribe((account) => {
             this.updateForm(account);
         });
+    }
+
+    save(): void {
+        this.success = false;
+
+        this.account.firstName = getFormControlUnchecked(this.settingsForm, 'firstName').value;
+        this.account.lastName = getFormControlUnchecked(this.settingsForm, 'lastName').value;
+        this.account.email = getFormControlUnchecked(this.settingsForm, 'email').value;
         <%_ if (enableTranslation) { _%>
-        this.languages = this.languageHelper.getAll();
+        this.account.langKey = getFormControlUnchecked(this.settingsForm, 'langKey').value;
         <%_ } _%>
 
-    }
+        this.accountService.save(this.account).subscribe(() => {
+            this.success = true;
 
-    save() {
-        const settingsAccount = this.accountFromForm();
-        this.accountService.save(settingsAccount).subscribe(() => {
-            this.error = null;
-            this.success = 'OK';
             this.accountService.identity(true).subscribe((account) => {
                 this.updateForm(account);
             });
             <%_ if (enableTranslation) { _%>
+
             this.languageService.getCurrent().then((current) => {
-                if (settingsAccount.langKey !== current) {
-                    this.languageService.changeLanguage(settingsAccount.langKey);
+                if (this.account.langKey !== current) {
+                    this.languageService.changeLanguage(this.account.langKey);
                 }
             });
             <%_ } _%>
-        }, () => {
-            this.success = null;
-            this.error = 'ERROR';
```

Comment:
Shouldn't we also set success to false and error to true when an error occurs from the http call ?

Reply:
Variable `error` was not used, so I removed this. Variable `success` is set to `false` at the beginning of this function. Errors are shown by `ErrorHandlerInterceptor`.  So this block is not needed.